### PR TITLE
Image Web/Desktop: Add support for http headers

### PR DESCRIPTION
**Pull request by kidroca**
### Details

This PR refactors the `Image` component, aligning the web/desktop and native implementations for better consistency and maintainability.

#### **Changes:**

1. **Introduced `BaseImage` Component:** 
    - File Location: `src/components/Image/BaseImage.js`
    - This new component acts as a bridge between the native and web implementations. 
    - For the `onLoad` event, it ensures that both web and native platforms have the same signature.

2. **Native Implementation Update:** 
    - File Location: `src/components/Image/BaseImage.native.js`
    - The native side now directly uses `react-native-fast-image`.

3. **Unified Image Component for Web/Desktop and Native:**
    - File Location: `src/components/Image/index.js`
    - The main `Image` component has been refactored to use the new `BaseImage`.
    - This change streamlines the authToken inclusion process in the image source, ensuring it works seamlessly across both platforms.

4. **Removed Redundant Native Specific Implementation:** 
    - File Location: `src/components/Image/index.native.js`
    - With the new unified approach, the native-specific implementation has become redundant and has been removed.

5. **Patch for `react-native-web`**
   - File Location: `patches/react-native-web+0.19.9+005+image-header-support.patch`
   - Implemented a custom patch for the official react-native-web version 0.19.9, incorporating changes from PR [#2442](https://github.com/necolas/react-native-web/pull/2442). This patch adds support for Image with HTTP headers, which was previously missing in our current version.

### Fixed Issues

$ https://github.com/Expensify/App/issues/12603
PROPOSAL: N/A

#### Verify the Image component works on all platforms
1. Launch the application and navigate to a report containing multiple attachments.
2. Scroll through the report, ensuring that all attachments are loading successfully without any errors or delay.
3. Click on an attachment to open its preview.
   - Confirm that the preview displays the image correctly.
   - Ensure the download button is present and functional; downloading the image should work as expected.
4. Open your browser's developer tools or the equivalent network monitoring tool for the platform you are testing on.
   - Monitor the network requests for image attachments.
   - Verify that the requests contain the `X-Chat-Attachment-Token` header.
5. Test the image caching functionality:
   - Keep the application open for more than 2 hours.
   - Observe that the images remain loaded and are not re-fetched.
   - Logout from the application and then log back in.
   - Navigate to the same report with attachments.
   - Images that were previously loaded should display instantly, indicating they are loaded from the cache.
   - Confirm that the behavior remains consistent even if the `X-Chat-Attachment-Token` value changes.

### Offline tests

1. Make sure you have viewed some reports with attachments while online.
2. Disconnect from the internet or turn on airplane mode to simulate an offline state.
3. Navigate to a report with attachments that were previously loaded.
   - Verify that these images still render correctly, even without an internet connection.
4. Navigate to a report with attachments that were NOT previously loaded.
   - The attachments should appear in an infinite loading state, indicating they cannot be fetched without an internet connection.

### QA Steps

1. Open the staging environment of the application.
2. Navigate to various reports that contain different types of attachments.
3. Ensure that all attachments load successfully without any visual artifacts or errors.
4. Click on any attachment to initiate the attachment preview.
   - Confirm that the preview opens, displaying the attachment correctly.
   - Locate the download button and click on it. Verify that the image is downloaded to your device or computer as expected.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/12156624/41d4f183-565d-44d6-91a4-8f508fccbb56

</details>

<details>
<summary>Mobile Web - Chrome</summary>

image

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-11-06 at 10 33 33

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/12156624/389fe185-d302-4153-9c0a-a95714387893

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

image

</details>

### CODE REVIEW

**Comment by Beamanator** [comment]
Holding on this as we try to get the upstream `react-native-web` PR merged: https://github.com/necolas/react-native-web/pull/2442

**Comment by kidroca** [comment]
Image requests made with headers, then the same requests getting served from cache
image

Still served from browser cache after login out, closing browser window, then opening a new window and loging in
image

Sample request content 
image
- no longer uses query param for the token
- has a `X-Chat-Attachment-Token` request header instead

**Comment by kidroca** [comment]
I've cherry picked this commit in order to test the new Image functionality: 34d941d92742dda5e383a19b759ef60247e5376a
Once the code makes it to `main` I'll drop the commit and merge `main` to my branch and conflicts should be resolved

Details about the related PR here: https://github.com/Expensify/App/issues/12603#issuecomment-1361399408

**Comment by Beamanator** [comment]
Thanks for this @kidroca!

> Once the code makes it to main

Note: There's a draft PR to re-implement the iOS / Android caching: https://github.com/Expensify/App/pull/13304 but we're not planning to move forward with that for another week (assuming the maintainer of https://github.com/DylanVann/react-native-fast-image/ doesn't respond sooner)

So if we get your `react-native-web` PR (Expensify fork) merged sooner, we may need to decide if we want to wait on that ^ draft PR or implement some of it in your PR to take advantage of your `react-native-web` changes

**Review by kidroca — COMMENTED** [review]
Added some notes on the changes

This PR is now based on the changes made about Fast-Image: https://github.com/Expensify/App/pull/13304

File: src/components/Image/BaseImage.js
Comment (Beamanator) [reply src/components/Image/BaseImage.js:21]: you're talking about `imageLoadedSuccessfully` would get redefined each render, right?

Is that a problem? It seems pretty lightweight, and I don't imagine images to re-render often, am I crazy?

File: src/libs/replaceSourceOrigin.js
Comment (Beamanator) [inline src/libs/replaceSourceOrigin.js:13]:
```suggestion
    const originsToReplace = [Config.EXPENSIFY.EXPENSIFY_URL, Config.EXPENSIFY.STAGING_EXPENSIFY_URL];
```

File: src/libs/replaceSourceOrigin.js
Comment (Beamanator) [inline src/libs/replaceSourceOrigin.js:8]: It looks like we always return a string, no?

```suggestion
 * @returns {String}
```

File: src/libs/replaceSourceOrigin.js
Comment (Beamanator) [inline src/libs/replaceSourceOrigin.js:10]: To be honest I think I'd rather see all of these changes in a different PR... but if this makes it impossible for you to test, maybe we can keep them here but we should mention you're trying to also fix https://github.com/Expensify/App/issues/14227 right?

**Review by Beamanator — COMMENTED** [review]
Love what I've seen so far, got a few initial comments

File: src/components/Image/BaseImage.js
Comment (kidroca) [reply src/components/Image/BaseImage.js:21]: Actually I haven't seen an example of such usage in App's code 

If we go with a functional component it would look like this

```jsx
const BaseImage = (props) => {
  const imageLoadedSuccessfully = ({ nativeEvent }) => {
    // We override `onLoad` to match our `propTypes.onLoad` interface
    const {width, height} = nativeEvent.source;
    props.onLoad({nativeEvent: {width, height}});
  }

  return <RNImage {...props} onLoad={imageLoadedSuccessfully} />;
}
```

Or 

```jsx
return (
  <RNImage {...props} 
    onLoad={({ nativeEvent }) => {
      // We override `onLoad` to match our `propTypes.onLoad` interface
      const {width, height} = nativeEvent.source;
      props.onLoad({nativeEvent: {width, height}});
  }} />
);
```

I think we use class components and `this.imageLoadedSuccessfully = this.imageLoadedSuccessfully.bind(this);` to get away from inline functions (even though here it would keep things simple)

File: src/libs/replaceSourceOrigin.js
Comment (kidroca) [reply src/libs/replaceSourceOrigin.js:10]: > To be honest I think I'd rather see all of these changes in a different PR... but if this makes it impossible for you to test, maybe we can keep them here but we should mention you're trying to also fix #14227 right?

I'm not exactly sure how this issue will be resolved
- #14227 

I might drop the changes or submit a separate PR if it turns out to be the accepted solution
In the meantime I keep the fix here to avoid reporting it as a problem of (and the react-native-web PR)
- #12603

Comment (kidroca) [reply src/libs/replaceSourceOrigin.js:10]: It won't make the PR impossible to test, but people will get CORS errors (on DEV) for images that were uploaded on staging

File: src/components/Image/BaseImage.js
Comment (kidroca) [reply src/components/Image/BaseImage.js:29]: > I'm down with that, though one solution could be:
> 
> * Change `imagePropTypes` to `baseImagePropTypes`
> * In the `Image` component props, spread `baseImagePropTypes` and the other missing props (`isAuthTokenRequired`, `session`)

Thanks for the idea, I can do that, but I see another inconsistency

`baseImagePropTypes` would only be used here in `BaseImage.js`, and not in `BaseImage.native.js`, so if they aren't shared by `BaseImage.*` and are only meant to be used in `BaseImage.js` then props should just be defined in `BaseImage.js`

**Comment by kidroca** [comment]
The changes about [src/libs/replaceSourceOrigin.js](https://github.com/Expensify/App/pull/13036/files#diff-9539f0f016eff48a88995c4d2e8c37bd80065f4404cee1ddc6f40f8a0e58d3c9) are from a separate PR (https://github.com/Expensify/App/pull/14463) and are carried here only for compatibility 
(I think the other PR would be merged first, and these changes would disappear from this PR)

**Comment by Beamanator** [comment]
@aimane-chnaif FYI the changes in the related fork PR are very close to being approved, so feel free to jump in and start testing this when you have a chance 👍 🙏

File: src/components/Image/BaseImage.js
Comment (Beamanator) [reply src/components/Image/BaseImage.js:21]: I agree we wouldn't want to inline, but I don't see a problem with the first version of the functional component 🤔

File: src/components/Image/BaseImage.js
Comment (Beamanator) [reply src/components/Image/BaseImage.js:29]: Good point, I like your reasoning 👍

File: src/components/Image/BaseImage.js
Comment (kidroca) [reply src/components/Image/BaseImage.js:21]: Cool, I'll update it once the react-native-web PR is done

**Comment by Beamanator** [comment]
@kidroca ok it looks like `@expensify/react-native-web` [`v0.18.11`](https://www.npmjs.com/package/@expensify/react-native-web/v/0.18.11) has been published and is ready for testing in this PR 👍

**Comment by kidroca** [comment]
Cool, I'll switch over to the new version and post back shortly

**Comment by kidroca** [comment]
I've switched the PR to the new react-native-web library 

I confirm requests are now made with a `X-Chat-Attachment-Token` header
So far this is working fine in DEV, but I think I found a CORS issue that would happen if we publish this to staging/prod

When we're on DEV we have a `web-proxy` that satisfies CORS rules
When we're on PROD the origin is new.expensify.com but ROOT_API is www.expensify.com so when we make a `fetch` request the browser would make access control checks

This is what I'm trying:
- make a production build for web: `npm run build`
- start a live server with the build `npx serve -s ./dist -p 3000`
- edit `/etc/hosts` and add an entry for `127.0.0.1       new.expensify.com` (make `new.expensify.com` resolve from my local server)
- start an https nginx server that relays requests to port 3000 
- clear Safari history (so that it forgets new.expensify.com otherwise it would not resolve from localhost)
- see CORS errors in console

So far this only works in Safari, but Safari lacks the full information in dev tools
Chrome is too clever - it does not allow to load new.expensify.com because the https certificate does not match

Is there a way to verify this would work before deploying to staging?

**Comment by kidroca** [comment]
Posted more info here: https://github.com/Expensify/App/issues/14227#issuecomment-1406800255

The short version is that because of the request header and the different origins, browsers would make a preflight OPTIONS request, before the actual GET request. If the OPTIONS request is not resolved correctly browsers would not make the GET request

Sample preflight request
```
curl -X OPTIONS \
https://www.expensify.com/chat-attachments/4198796077944938521/w_6f21f50b3af2ee93c4e6e288ba5b986858166367.jpg.1024.jpg \
--dump-header -
```

Right now the response is 302 (not 200) and browsers aren't proceeding with the GET request

**Comment by kidroca** [comment]
Resolved package conflicts

**Comment by Beamanator** [comment]
@aimane-chnaif are you able to review this soon? 😅

**Comment by Beamanator** [comment]
@kidroca would you please write testing steps in the OP and complete the author checklist?

**Comment by kidroca** [comment]
@Beamanator 
I think I'll have to make an update to the `desktop/main.js` code in order to handle CORS
We're getting the same problem described here: https://github.com/Expensify/App/pull/13036#issuecomment-1406805005 
Electron / Chromium tries to make OPTIONS request before fetching an image with headers

Can someone confirm what's the plan regarding the CORS issue:
I think the plan is to configure the API server to allow CORS requests originating from `new.expensify` and `staging.new.expensify`
So I think to fix the Desktop issue I'll have to do the same

The other way around this would be to have a proxy (similar to how we have one for local development). I've shared it here: https://github.com/Expensify/App/issues/14227#issuecomment-1410089720, but I'm not sure what's decided in the end

So in order to continue I need to find out how we'll deal with the CORS issue

**Comment by kidroca** [comment]
@aimane-chnaif 
I'm not sure, but it seems you're intentionally altering code to make invalid header calls 

When would `headers` be `undefined` or `null` and produce the error from the video? 
If the token expires it should automatically update and this should cause images to reload (due to the heder change)
Also if an image is recently loaded it would resolve from cache even if the header expired (or changed) 

It seems `headers` would be `null` if `'session.encryptedAuthToken'` is missing (no value in Onyx)
But this should logout the user and they would be on the login screen where we don't try to load images using auth

---

I can't tell what causes the request to repeat, it might be the backend response or the way the Image renders internally (existing logic that makes requests without headers is unchanged) or the way you've setup code to remove headers from the request

**Comment by kidroca** [comment]
We can simplify the code here:

```js
    if (typeof source !== 'number' && isAuthTokenRequired) {
        const authToken = lodashGet(props, 'session.encryptedAuthToken', null);
        imageSource = {
            ...source,
            headers: authToken ? {
                [CONST.CHAT_ATTACHMENT_TOKEN_KEY]: authToken,
            } : null,
        };
    }
```

To:
```js
    if (typeof source !== 'number' && isAuthTokenRequired) {
        const authToken = lodashGet(props, 'session.encryptedAuthToken');
        imageSource = {
            ...source,
            headers: {
                [CONST.CHAT_ATTACHMENT_TOKEN_KEY]: authToken,
            },
        };
    }
```

This should result in the "invalid token" flow, if it's ever the case that the token is missing in Onyx and at the same time we're trying to load an image that requires auth

File: package.json
Comment (Beamanator) [inline package.json:44]: We recently published v0.18.12 of our fork, can you update so we're testing on the latest package? 🙏

**Comment by kidroca** [comment]
> @kidroca I don't think we should only discuss the CORS issue here, I think we need to bring this to slack to get more 👀 . Do you have a convo going about this in another issue / PR / slack thread already?

I think the CORS issue is supposed to be resolved here: https://github.com/Expensify/App/issues/14227#issuecomment-1410089720
I'm waiting for an update: https://github.com/Expensify/App/issues/14227#issuecomment-1416052221

For now we should make a note to not merge this PR until we've addressed the CORS issue 
_(I'll fix the merge conflicts now)_

**Comment by kidroca** [comment]
I've fixed the merge conflict, but I can't get the Desktop app to work, because Electron/Chromium makes a preflight request 

1. Electron sees a request with a `X-Chat-Attachment-Token` header
2. Electron makes a Preflight (OPTIONS method) request to `https://www.expensify.com/chat-attachments/...` to ask the server whether it can make this request for the current origin and what headers are allowed (e.g. is the token header allowed)
3. Backend responds to the Preflight (OPTIONS) request with a redirect
4. Electron aborts the request because a redirect is invalid response for a preflight request

Here's a sample error that I'm getting in Electron

```
Access to fetch at 'https://www.expensify.com/chat-attachments/962115596/w_e989bfe450cb8e872f52dac2d082c391f35e2e0a.jpg.1024.jpg' from origin 'http://localhost:8081' 
has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.
```

I've tried to tweak `desktop/main.js` code to alter the origin/referer in order to bypass cors, the way we do for PROD, but it doesn't seem to work
https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/desktop/main.js#L298-L305

- I've removed `if (!__DEV__)` but it doesn't seem to change the origin or the referrer of the actual request

**Comment by kidroca** [comment]
I think to continue here we first need to deal with the CORS issue and the OPTIONS request 
Decide how to setup the backend to work 
- setup CORS so that OPTIONS requests are addressed correctly
- or make these non-CORS requests by making the requests against the host origin instead of API_ROOT

**Comment by Beamanator** [comment]
Thanks for all of your work on this @kidroca - looks like we'll continue this CORS conversation in https://github.com/Expensify/App/issues/14991 👍

**Review by kidroca — COMMENTED** [review]
@aimane-chnaif @Beamanator

I've synchronized with `main` and resolved the merge conflicts.
I can confirm that the Image loading using the auth header is functional on web/desktop.
I've added some review notes to provide context and facilitate the PR review process.

The PR is now ready for review.

**Comment by Beamanator** [comment]
@aimane-chnaif are you able to review this soon? If not, can we quickly assign another C+ to help so we can get this movin? :D

Also @kidroca conflicts, sorry

**Comment by kidroca** [comment]
I have brought this PR up to date with `main`. 

Additionally, I've created a patch via `patch-package` that incorporates the enhancements from [this pull request](https://github.com/necolas/react-native-web/pull/2442) in `react-native-web`.
- These changes are planned for `react-native-web` version 0.20. However, since our project currently uses version 0.19.9, which does not yet include these updates, applying a patch was necessary to integrate the new features.
- _more details as why we have to use a patch are available on [this and following comments](https://github.com/Expensify/App/issues/12603#issuecomment-1804450624)_

I conducted thorough retesting on both web and desktop platforms. I can confirm that images are now successfully loading through requests that include authentication in their headers, as per this PR goal.

**Comment by kidroca** [comment]
Discovered and fixed an issue with Avatars opened for preview in the `AttachmentModal`
- the `isAuthTokenRequired` was used on the AttachmentModal used on avatars, since avatars are no loaded from the same origin as app, making a request with our custom auth token is unexpected by the end point and resulted in CORS error
- more details in this commit: 061ee6ea608bee0c33ff95c2aef381b93034e73a

**Comment by Beamanator** [comment]
Thanks @kidroca - I posted in that PR as well, I don't see / remember why we ever needed auth tokens to fetch avatars.

It sounds like your PR is ready to test again, right? 🙏

**Comment by Beamanator** [comment]
Amazing, thanks @kidroca ! 

@aimane-chnaif are you available to test? 🙏

**Comment by kidroca** [comment]
> @kidroca please merge main

Done

File: src/components/Image/index.native.js
Comment (kidroca) [reply src/components/Image/index.native.js:1]: It was confirmed as leftover

**Review by aimane-chnaif — COMMENTED** [review]
LGTM! Some minor feedback

**Comment by kidroca** [comment]
Pushed a commit to address the code style issues

**Comment by kidroca** [comment]
> can you please also pull main and retest? almost 1k commits behind

I've synced and re-tested - everything seems good, though there seems to be a Typescript error unrelated to my changes

```console
src/libs/actions/Link.ts:103:29 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'AllRoutes | undefined'.
```

**Comment by kidroca** [comment]
> @kidroca please merge main again. It's fixed in #32510

✔️ Done

**Review by aimane-chnaif — APPROVED** [review]
🎉

**Review by Beamanator — APPROVED** [review]
LGTM!

**Pull request merged**
